cns/service/main.go

```diff
@@ -536,6 +537,8 @@ func main() {
 		switch scenario := cniConflistScenario(scenarioString); scenario {
 		case scenarioV4Overlay:
 			conflistGenerator = &cniconflist.V4OverlayGenerator{Writer: writer}
+		case scenarioOverlay:
+			conflistGenerator = &cniconflist.OverlayGenerator{Writer: writer}
```

Comment:
should we be replacing the existing v4overlay generator with this new one?

Reply:
As a follow up to this PR, I believe we need to make change in agent baker test files to reflect the latest mode ( not making that change won't break anything AFAIK but it's just to reflect the up-to-date file contents )

Reply:
> should we be replacing the existing v4overlay generator with this new one?

Yeah I think that would be best, I thought Matt wanted to keep it in case of roll back
@thatmattlong do you want me to keep this v4overlay value or remove it?

Reply:
Hey @thatmattlong from what I can see in AKS-RP, there is nowhere this feature (CNS generating cni conflist for overlay) is being used, is that correct?
In that case, I could remove the "v4overlay" value in this PR safely

Reply:
Nvm, aks DOES use the conflist generator feature of CNS, so we'd leave 'v4overlay' as an option